### PR TITLE
Fix and enable test_line_info.py

### PR DESCRIPTION
**Pull request by etiotto**
Enable `test_line_info.py` to test that line number information are emitted in the SPIRV binary generated by the compiler.

### CODE REVIEW

File: python/test/unit/language/test_line_info.py [hunk 1]
```diff
@@ -109,8 +140,9 @@ def check_file_lines(file_lines, file_name, lineno, should_contain=True):
     return not should_contain
 
 
-func_types = ["single", "call", "call_noinline", "multi_files", "autotune", "dot_combine"]
-
+#TODO: dot_combine fails to compile.
```
Comment (etiotto) [inline python/test/unit/language/test_line_info.py:143]: Problem tracked by issue https://github.com/intel/intel-xpu-backend-for-triton/issues/191

File: python/test/unit/language/test_line_info.py [hunk 2]
```diff
@@ -109,15 +125,16 @@ def check_file_lines(file_lines, file_name, lineno, should_contain=True):
...
+#func_types = ["single", "call", "call_noinline", "multi_files", "autotune", "dot_combine"]
```
Comment (whitneywhtsang) [inline python/test/unit/language/test_line_info.py:129]: [nit]
```suggestion
# TODO: dot_combine fails to compile.
# func_types = ["single", "call", "call_noinline", "multi_files", "autotune", "dot_combine"]
```

File: python/test/unit/language/test_line_info.py [hunk 3]
```diff
@@ -154,7 +172,6 @@ def test_line_info(func: str):
         assert (check_file_lines(file_lines, "standard.py", 34))
         assert (check_file_lines(file_lines, "standard.py", 36))
     elif func == "autotune":
-        assert (check_file_lines(file_lines, "test_line_info.py", 60))
```
Comment (whitneywhtsang) [inline python/test/unit/language/test_line_info.py:157]: Why this assert is removed?

Comment (etiotto) [reply python/test/unit/language/test_line_info.py:157]: No such line number in the SPV binary.

Comment (whitneywhtsang) [reply python/test/unit/language/test_line_info.py:157]: ok.

**Comment by etiotto** [comment]
@whitneywhtsang are you OK with this PR now ?

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**